openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceGenerationUtils.java

```diff
@@ -275,6 +264,26 @@ public static ImmutablePair<Optional<TypeDescriptorNode>, Optional<TypeDefinitio
         }
     }
 
+    /**
+     *
+     * This util uses for selecting standard media type by looking at the user defined media type.
+     */
+    public static String seleteMediaType(String mediaTypeContent) {
+
+        if (mediaTypeContent.matches("text/.*")) {
+            mediaTypeContent = GeneratorConstants.TEXT;
+        } else if (mediaTypeContent.matches("application/.*\\+json")) {
+            mediaTypeContent = GeneratorConstants.APPLICATION_JSON;
+        } else if (mediaTypeContent.matches("application/.*\\+xml")) {
+            mediaTypeContent = GeneratorConstants.APPLICATION_XML;
+        } else if (mediaTypeContent.matches("application/.*\\+octet-stream")) {
+            mediaTypeContent = GeneratorConstants.APPLICATION_OCTET_STREAM;
+        } else if (mediaTypeContent.matches("application/.*\\+x-www-form-urlencoded")) {
+            mediaTypeContent = GeneratorConstants.APPLICATION_URL_ENCODE;
+        }
+        return mediaTypeContent;
+    }
```

Comment:
whats the expected output for the media types like `text/json`, `text/xml`, etc?

Reply:
Currently, the tool was mapping with the string type. will re-change the util as per discussed in the client generation media type mapping. `text/json ` for JSON and `text/xml` for XML.